src/mono/wasm/runtime/driver.c

```diff
@@ -969,6 +969,27 @@ mono_wasm_get_type_aqn (MonoType * typePtr) {
 	return mono_type_get_name_full (typePtr, MONO_TYPE_NAME_FORMAT_ASSEMBLY_QUALIFIED);
 }
 
+// this will return bool value if the object is a bool, otherwise it will return -1 or error
+EMSCRIPTEN_KEEPALIVE int
+mono_wasm_read_as_bool_or_null_unsafe (PVOLATILE(MonoObject) obj) {
+	MONO_ENTER_GC_UNSAFE;
+
+	MonoClass *klass = mono_object_get_class (obj);
+	if (!klass) return -1;
+
+	MonoType *type = mono_class_get_type (klass);
+	if (!type) return -1;
+
+	int mono_type = mono_type_get_type (type);
+	if (MONO_TYPE_BOOLEAN == mono_type) {
+		return ((signed char*)mono_object_unbox (obj) == 0 ? 0 : 1);
+	}
+
+	return -1;
+
+	MONO_EXIT_GC_UNSAFE;
```

Comment:
is it correct to exit the gc unsafe region after a return statement? from reading the definition of the macro I don't think it is.

Reply:
fixed, thanks!!